packages/ui/src/proposals/components/ProposalDiscussions.tsx

```diff
@@ -43,23 +55,21 @@ export const ProposalDiscussions = ({ thread, proposalId }: Props) => {
       return api.tx.proposalsDiscussion.addPost(
         createType('MemberId', Number.parseInt(active.id)),
         thread.id,
-        metadataToBytes(ForumPostMetadata, { text: postText }),
+        metadataToBytes(ForumPostMetadata, { text: postText, repliesTo: replyTo ? Number(replyTo.id) : undefined }),
         isEditable
       )
     }
   }
 
-  const getReplyComponent = () => {
-    if (thread.mode === 'open') {
-      return <NewThreadPost getTransaction={getTransaction} />
+  const getPostForm = () => {
+    if (thread.mode === 'open' || (thread.mode === 'closed' && active && thread.whitelistIds?.includes(active.id))) {
```

Comment:
Either a callback as `isAbleToPost()` or a variable `isAbleToPost` - it will be easier to parse later ;)